### ISSUE TITLE
Not working for androids now

### ISSUE THREAD

**Issue by DEEPAK-MSV** [taniabezancon/Pedometer-App-react-native#1]
This code doesn’t work for android devices

**Comment by taniabezancon** [comment]
thanks, you're right I should have mention it.

**Comment by DEEPAK-MSV** [comment]
> thanks, you're right I should have mention it.

You can give permission and use component to count steps for android

**Comment by sultan-arshi** [comment]
is it working for android ?
if yes, can you please guide me

**Comment by DEEPAK-MSV** [comment]
> is it working for android ? if yes, can you please guide me

yes it is working for Androids but you have to write Activity Recognition Permission to get the data from the sensors


 const permission =()=>{
      PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.ACTIVITY_RECOGNITION) 
    }  
 
    useEffect(()=>{
      permission();
    },[]);


    useEffect(() => {
      // Get the saved daily step count on mount
      const getSavedStepCount = async () => {
        const savedCount = await AsyncStorage.getItem('dailyStepCount');
        if (savedCount) {
          SetSteps(parseInt(savedCount));
        }
      };
  
      // Subscribe to updates in step count
      const subscription = Pedometer.watchStepCount(result => {
        SetSteps(result.steps);
        // Save the updated daily step count to AsyncStorage
        AsyncStorage.setItem('dailyStepCount', result.steps.toString());
      });
      const resetStepCount = () => {
        setDailyStepCount(0);
        AsyncStorage.setItem('dailyStepCount', '0');
      };
      const now = new Date();
      const midnight = new Date(
        now.getFullYear(),
        now.getMonth(),
        now.getDate() + 1,
        0, 0, 0
      );
      const timeUntilMidnight = midnight.getTime() - now.getTime();
      const intervalId = setInterval(resetStepCount, timeUntilMidnight);
  
      // Update the daily step count on mount
      getSavedStepCount();
  
      // Clean up subscription and interval on unmount
      return () => {
        subscription.remove();
        clearInterval(intervalId);
      };
    }, []);
 use this as reference
